src/main/java/com/revature/sfadapter/services/SFAuthService.java

```diff
@@ -1,86 +1,93 @@
+
+		return url.toString();
 	}
 
-	private String getRequestUrl(String clientRedirect){
-		String clientId = System.getenv("SF_CONSUMER_KEY");
+	private MultiValueMap<String, String> buildAccessRequestFormData(String accessCode){
+		MultiValueMap<String, String> data = new LinkedMultiValueMap<>();
 
-		String url = "https://test.salesforce.com/services/oauth2/authorize?response_type=code";
-		url += "&client_id=" + clientId;
-		url += "&redirect_uri=" + URLEncoder.encode("https://sf.aduet.tech/services/authenticate");
-		url+= "&state=" + URLEncoder.encode(clientRedirect);
-		url += "&prompt=login";
+		data.add("grant_type", "authorization_code");
+		data.add("code", accessCode);
+		data.add("client_id", clientId);
+		data.add("client_secret", clientSecret);
+		data.add("redirect_uri",myCallbackUrl);
 
-		return url;
+		return data;
 	}
 
-	public void sendForAuth(HttpServletResponse response, String clientRedirect) throws IOException{
-		String url = getRequestUrl(clientRedirect);
-		response.sendRedirect(url);
+	public String getAuthUrl(String clientRedirect) throws IOException{
+		LOGGER.debug("Building url for sales force request");
+		String authUrl = buildAuthorizationUrl("/services/oauth2/authorize", clientRedirect);
+		LOGGER.error("getAuthUrl is incomplete");
+		return null;
 	}
 
 	public SFWSAccessObject sendForAccess(String accessCode){
-		System.out.println("Starting rest post to Salesforce");
-		MultiValueMap<String, String> data = new LinkedMultiValueMap<>();
+
 		SFWSAccessObject accessObj;
 		RestTemplate client = new RestTemplate();
 
-		String clientId = System.getenv("SF_CONSUMER_KEY");
-		String clientSecret = System.getenv("SF_CONSUMER_SECRET");
+		LOGGER.debug("Preparing request for Salesforce access token");
+		LOGGER.debug("Building form data for request");
+		MultiValueMap<String, String>data = buildAccessRequestFormData(accessCode);
 
-		data.add("grant_type", "authorization_code");
-		data.add("code", accessCode);
-		data.add("client_id", clientId);
-		data.add("client_secret", clientSecret);
-		data.add("redirect_uri","https://sf.aduet.tech/services/authenticate");
+		Assert.isTrue(data.size() > 0);
 
 		HttpHeaders headers = new HttpHeaders();
 		headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
 		headers.setAccept(Arrays.asList(MediaType.APPLICATION_JSON_UTF8));
 		HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<>(data, headers);
 		System.out.println(request.toString());
-		System.out.println("Post ready sending to Salesforce");
-		ResponseEntity<SFWSAccessObject> obj = client.postForEntity("https://test.salesforce.com/services/oauth2/token", request, SFWSAccessObject.class);
-		return obj.getBody();
 
-	}
+		LOGGER.debug("Request ready sending to Salesforce");
+		LOGGER.error("sendForAccess is incomplete");
+		ResponseEntity<SFWSAccessObject> obj = client.postForEntity(authorizationUrl, request, SFWSAccessObject.class);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "obj" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revatureops.com/coding_rules#rule_key=squid%3AS1481)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "obj". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revatureops.com/coding_rules#rule_key=squid%3AS1854)